File: RelacionamentoClientes/src/br/com/inatel/pos/dm102/bo/AtendimentoBO.java
package br.com.inatel.pos.dm102.bo;

import java.util.ArrayList;
import java.util.Date;
import java.util.Scanner;

import br.com.inatel.pos.dm102.model.Atendimento;
import br.com.inatel.pos.dm102.model.Cliente;
import br.com.inatel.pos.dm102.model.PessoaFisica;
import br.com.inatel.pos.dm102.model.PessoaJuridica;

public class AtendimentoBO {

	private ArrayList<Atendimento> atendimentos = new ArrayList<>();

	public void iniciarAtendimento() {

		do {

			System.out.println("------ Relacionamento de Clientes -------");

			System.out.println("> 1 - Novo Atendimento");
			System.out.println("> 2 - Listar Atendimentos");
			System.out.println("\n> 3 - Sair da aplicação");

			Scanner in = new Scanner(System.in);
			String op = in.nextLine();

			switch (op) {
			case "1":
				comecarAtendimento();

				break;
			case "2":
				listarAtendimentos();

				break;
			case "3":
				System.exit(0);
				break;

			default:
				break;
			}
			in.close();
		} while (true);

	}

	private void listarAtendimentos() {

		System.out.println("Lista de atendimentos");

		for (Atendimento atendimento : atendimentos) {

			System.out.println(atendimento.toString());
		}
		System.out.println("\n\n");

	}

	private void comecarAtendimento() {
		System.out.println("> Tipo de Pessoa");
		System.out.println("> 1 - Pessoa Fisica");
		System.out.println("> 2 - Pessoa Juridica");

		System.out.println("> 3 - Sair da aplicação");

		System.out.println("Escolha o tipo de pessoa: ");

		Scanner in = new Scanner(System.in);
		String op = in.nextLine();

		switch (op) {
		case "1":
			PessoaFisica pessoaFisica = new PessoaFisica();
			System.out.println(">> Pessoa Fisica");
			System.out.println(">> Entre com o nome:");
			pessoaFisica.setNome(in.nextLine());
			System.out.println(">> Entre com o endereço:");
			pessoaFisica.setEndereço(in.nextLine());
			System.out.println(">> Entre com o telefone:");
			pessoaFisica.setTelefone(in.nextLine());
			System.out.println(">> Entre com o cpf:");
			pessoaFisica.setCpf(in.nextLine());
			System.out.println(">> Entre com o identidade:");
			pessoaFisica.setIdentidade(in.nextLine());
			System.out.println(">> Entre com o tipo identidade:");
			pessoaFisica.setTipoIdentidade(in.nextLine());

			criarAtendimento(pessoaFisica);

			break;
		case "2":
			PessoaJuridica pessoaJuridica = new PessoaJuridica();
			System.out.println(">> Pessoa Juridica");
			System.out.println(">> Entre com o nome:");
			pessoaJuridica.setNome(in.nextLine());
			System.out.println(">> Entre com o endereço:");
			pessoaJuridica.setEndereço(in.nextLine());
			System.out.println(">> Entre com o telefone:");
			pessoaJuridica.setTelefone(in.nextLine());
			System.out.println(">> Entre com o cnpj:");
			pessoaJuridica.setCnpj(in.nextLine());
			System.out.println(">> Entre com o razao social:");
			pessoaJuridica.setRazaoSocial(in.nextLine());
			System.out.println(">> Entre com a inscrição estadual:");
			pessoaJuridica.setInscricaoEstadual(in.nextLine());

			criarAtendimento(pessoaJuridica);

			break;
		default:
			break;
		}
		in.close();
	}

	private void criarAtendimento(Cliente cliente) {

		Scanner in = new Scanner(System.in);

		System.out.println(">> Criar atendimento");

		Atendimento atendimento = new Atendimento();
		atendimento.setCliente(cliente);
		atendimento.setData(new Date());

		System.out.println(">> Insira uma descrição para o atendimento: ");

		atendimento.setDescricao(in.nextLine());

		atendimentos.add(atendimento);
		in.close();
		System.out.println("Atendimento criado com sucesso!");
		System.out.println("\n\n");
	}
}

File: RelacionamentoClientes-C#/RelacionamentoClientes/RelacionamentoClientes/Business/AtendimentoBO.cs
using RelacionamentoClientes.Model;
using System;
using System.Collections.Generic;

namespace RelacionamentoClientes.Business
{
    class AtendimentoBO
    {
        List<Atendimento> atendimentos = new List<Atendimento>();

        public void IniciarAtendimento()
        {
            
            do
            {

               Console.WriteLine("------ Relacionamento de Clientes -------");

                Console.WriteLine("> 1 - Novo Atendimento");
                Console.WriteLine("> 2 - Listar Atendimentos");
                Console.WriteLine("\n> 3 - Sair da aplicação");
                String op = Console.In.ReadLine();
                switch (op)
                {
                    case "1":
                        ComecarAtendimento();

                        break;
                    case "2":
                        ListarAtendimentos();

                        break;
                    case "3":
                        
                        break;
                }

            } while (true);

        }

        private void ListarAtendimentos()
        {

            Console.WriteLine("Lista de atendimentos");
            
            foreach (Atendimento atendimento in atendimentos)
            {

                Console.WriteLine(atendimento.toString());
            }
            Console.WriteLine("\n\n");

        }

        private void ComecarAtendimento()
        {
            Console.WriteLine("> Tipo de Pessoa");
            Console.WriteLine("> 1 - Pessoa Fisica");
            Console.WriteLine("> 2 - Pessoa Juridica");

            Console.WriteLine("> 3 - Sair da aplicação");

            Console.WriteLine("Escolha o tipo de pessoa: ");


            String op = Console.In.ReadLine();

            switch (op)
            {
                case "1":
                    PessoaFisica pessoaFisica = new PessoaFisica();
                    Console.WriteLine(">> Pessoa Fisica");
                    Console.WriteLine(">> Entre com o nome:");
                    pessoaFisica.setNome(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o endereço:");
                    pessoaFisica.setEndereço(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o telefone:");
                    pessoaFisica.setTelefone(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o cpf:");
                    pessoaFisica.setCpf(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o identidade:");
                    pessoaFisica.setIdentidade(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o tipo identidade:");
                    pessoaFisica.setTipoIdentidade(Console.In.ReadLine());

                   

                    CriarAtendimento(pessoaFisica);

                    break;
                case "2":
                    PessoaJuridica pessoaJuridica = new PessoaJuridica();
                    Console.WriteLine(">> Pessoa Juridica");
                    Console.WriteLine(">> Entre com o nome:");
                    pessoaJuridica.setNome(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o endereço:");
                    pessoaJuridica.setEndereço(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o telefone:");
                    pessoaJuridica.setTelefone(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o cnpj:");
                    pessoaJuridica.setCnpj(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com o razao social:");
                    pessoaJuridica.setRazaoSocial(Console.In.ReadLine());
                    Console.WriteLine(">> Entre com a inscrição estadual:");
                    pessoaJuridica.setInscricaoEstadual(Console.In.ReadLine());
                    

                    CriarAtendimento(pessoaJuridica);

                    break;
                default:
                    break;
            }
        }

        private void CriarAtendimento(Cliente cliente)
        {
            

            Console.WriteLine(">> Criar atendimento");

            Atendimento atendimento = new Atendimento();
            atendimento.setCliente(cliente);
            atendimento.setData(DateTime.Now);

            Console.WriteLine(">> Insira uma descrição para o atendimento: ");

            atendimento.setDescricao(Console.In.ReadLine());

            atendimentos.Add(atendimento);
            
            Console.WriteLine("Atendimento criado com sucesso!");
            Console.WriteLine("\n\n");
        }



    }
}
